caproto/ioc_examples/ternaryarray.py

```diff
@@ -0,0 +1,143 @@
+import asyncio
+
+from functools import partial
+from collections import OrderedDict
+from caproto.server import PVGroup, ioc_arg_parser, pvproperty, run
+from ophyd import EpicsSignal, EpicsSignalRO, PVPositionerPC
+from ophyd import Component as Cpt
+
+
+class TernaryDeviceSim:
+    """
+    A device with three states.
+
+    Parameters
+    ----------
+    delay: float, optional
+        The time it takes for the device to change from state-0 to state-1.
+    """
+
+    def __init__(self, delay=0.5):
+        self._delay = delay
+        self._state = None
+
+    async def set(self):
+        if not self._state:
+            self._state = None
+            await asyncio.sleep(self._delay)
+            self._state = True
+
+    async def reset(self):
+        if self._state:
+            self._state = None
+            await asyncio.sleep(self._delay)
+            self._state = False
+
+    @property
+    def state(self):
+        return self._state
+
+
+class TernaryArrayIOC(PVGroup):
+    """
+    Example IOC that has an array of TernaryDevices.
+
+    Parameters
+    ----------
+    count: integer
+        The number of devices in the array.
+    """
+
+    def __init__(self, count=10, *args, **kwargs):
+        self._devices = [TernaryDeviceSim() for i in range(count)]
+
+        # Dynamically setup the pvs.
+        for i in range(count):
+            # Create the setpoint pv.
+            setattr(
+                self, f"device{i}", pvproperty(value=0, dtype=int, name=f"device{i}")
```

Comment:
How does this translate {True, False, None} to an integer?

Reply:
I'm going back and forth a bit it figure out the best way to do this.

I was thinking that I would be able to make TernaryDevice general, so that it can work with any device that has the same behavior.  So that it doesn't make any assumptions about pv names or the strings that get returned by status pv.
Then I was planning on subclassing TernaryDevice, to make a class called Filter which includes some of specifics about the cms filters, like the strings that get returned by the status PV.
Then I was planning on making a device that has an array of Filters for the cms attenuator.

Anyway, I'll have to work out this detail before it is ready to merge.

Reply:
Can we use enums for this?